scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt

```diff
@@ -1037,19 +1039,39 @@ internal class ContainerFactory(
           // PROD-8074: In the AMQP Consumers, Shared Access Policy credentials take precedence over
           // the tenant client credentials => we can therefore safely append the former to the
           // existing env vars.
-          mapOf(
-              AZURE_EVENT_HUB_SHARED_ACCESS_POLICY_ENV_VAR to
-                  (eventBus.authentication.sharedAccessPolicy?.namespace?.name
-                      ?: throw IllegalStateException(
-                          "Missing configuration property: " +
-                              "csm.platform.azure.eventBus.authentication.sharedAccessPolicy" +
-                              ".namespace.name")),
-              AZURE_EVENT_HUB_SHARED_ACCESS_KEY_ENV_VAR to
-                  (eventBus.authentication.sharedAccessPolicy?.namespace?.key
-                      ?: throw IllegalStateException(
-                          "Missing configuration property: " +
-                              "csm.platform.azure.eventBus.authentication.sharedAccessPolicy" +
-                              ".namespace.key")))
+          // PROD-8599: AZURE_EVENT_HUB_SHARED_ACCESS_POLICY_ENV_VAR,
+          // AZURE_EVENT_HUB_SHARED_ACCESS_KEY_ENV_VAR
+          // are renamed, but are kept for backward compability reason
+          when (eventBus.usePasswordForAMQPConsumer) {
```

Comment:
For backward compatibility, I suggest you just append the new CSM_AMQPCONSUMER_* environment variables to the map containing the existing AZURE_EVENT_HUB_SHARED_ACCESS_* ones.